src/Symfony/Bridge/Twig/Tests/Extension/TranslationExtensionTest.php

```diff
@@ -148,6 +166,39 @@ public function testDefaultTranslationDomain()
         $this->assertEquals('foo (foo)foo (custom)foo (foo)foo (custom)foo (foo)foo (custom)', trim($template->render(array())));
     }
 
+    public function testDefaultTranslationDomainWithNamedArguments()
+    {
+        $templates = array(
+            'index' => '
+                {%- extends "base" %}
+
+                {%- trans_default_domain "foo" %}
+
+                {%- block content %}
+                    {{- "foo"|trans(arguments = {}, domain = "custom") }}
+                    {{- "foo"|transchoice(count = 1) }}
+                    {{- "foo"|transchoice(count = 1, arguments = {}, domain = "custom") }}
+                    {{- "foo"|trans({}, domain = "custom") }}
+                    {{- "foo"|transchoice(1, arguments = {}, domain = "custom") }}
```

Comment:
you have at least 1 missing case here (and this case is broken according to my code review): passing the domain as a positional argument while using a named argument later in the argument list (for the locale for instance)